src/main/java/tigerlily/Parser.java

```diff
@@ -0,0 +1,106 @@
+package tigerlily;
+
+import tigerlily.commands.ByeCommand;
+import tigerlily.commands.Command;
+import tigerlily.commands.DeadlineCommand;
+import tigerlily.commands.DeleteCommand;
+import tigerlily.commands.EventCommand;
+import tigerlily.commands.FindCommand;
+import tigerlily.commands.ListCommand;
+import tigerlily.commands.MarkCommand;
+import tigerlily.commands.ToDoCommand;
+import tigerlily.commands.UnmarkCommand;
+
+import tigerlily.exceptions.DukeExceptions;
+import tigerlily.exceptions.InvalidFilePathException;
+import tigerlily.exceptions.UnknownCommandException;
+
+import tigerlily.tasks.Deadline;
+import tigerlily.tasks.Event;
+import tigerlily.tasks.Task;
+import tigerlily.tasks.ToDo;
+
+import java.time.LocalDateTime;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+    public enum CommandType {BYE, LIST, TODO, DEADLINE, EVENT, MARK, UNMARK, DELETE, FIND}
+
+    /**
+     * Handles the commands given by a user and returns the appropriate command to be executed.
+     *
+     * @param input the input given by a user
+     * @return the required Command
+     * @throws UnknownCommandException If a given command doesn't match the supported command types
+     */
+    public Command handleCommand(String input) throws DukeExceptions {
+        try {
+            CommandType command = CommandType.valueOf(input.split(" ")[0].toUpperCase());
+            switch (command) {
+            case BYE:
+                return new ByeCommand();
+            case LIST:
+                return new ListCommand();
+            case TODO:
+                return new ToDoCommand(input);
+            case DEADLINE:
+                return new DeadlineCommand(input);
+            case EVENT:
+                return new EventCommand(input);
+            case MARK:
+                return new MarkCommand(input);
+            case UNMARK:
+                return new UnmarkCommand(input);
+            case DELETE:
+                return new DeleteCommand(input);
+            case FIND:
+                return new FindCommand(input);
+            default:
+                throw new UnknownCommandException();
+            }
+        } catch (IllegalArgumentException e) {
+            throw new UnknownCommandException();
+        }
+    }
```

Comment:
I like the use of a switch rather than multiple if/nested-if statements for ur handleCommand method.